bluemix/src/main/java/com/ibm/watson/SpeechToText.java

```diff
@@ -43,120 +39,157 @@ public String speechToText(byte[] incomingBinaryPOSTData) {
-	    JsonReader responseReader = Json.createReader(new StringReader(sb.toString()));
-	    JsonObject jsonResponse = responseReader.readObject();
-	    responseReader.close();
-
-	    String transcript = "";
-	    try {
-		JsonArray resultsArray = jsonResponse.getJsonArray("results");
-		
-		for (JsonValue jv : resultsArray) {
-		    JsonObject jo = (JsonObject) jv;
-		    transcript = transcript + jo.getJsonArray("alternatives")
-			.getJsonObject(0)
-			.getString("transcript");
-		}
-	    } catch(Exception e) {
-		e.printStackTrace();
-		return XML_DEC + "<results>" + STATUS_ERROR + "<statusInfo>Unexpected response from service caused " + e.getClass().getName() + ": " + e.getMessage() + "</statusInfo><transcript /></results>";
-	    }
-            
             // Return the response from the service
-            return XML_DEC + "<results>" + STATUS_OK + "<transcript>" + transcript + "</transcript></results>";
-
-        } catch(Exception e){
+            return successXML(transcript);
+        } catch (Exception e) {
             e.printStackTrace();
-            return XML_DEC + "<results>" + STATUS_ERROR + "<statusInfo>Bluemix " + e.getClass().getName() + ": " + e.getMessage() + "</statusInfo><transcript /></results>";
-        }finally {
+            return failXML(e, "Bluemix ");
+        } finally {
             try {
                 if (conn != null) {
                     conn.disconnect();
                 }
             } catch (Exception e) {
                 e.printStackTrace();
-                return XML_DEC + "<results>" + STATUS_ERROR + "<statusInfo>Bluemix " + e.getClass().getName() + ": " + e.getMessage() + "</statusInfo><transcript /></results>";
+                return failXML(e, "Bluemix ");
             }
         }
-        
+
+    }
+
+
+    // Set up a connection with the speech-to-text service
+    private HttpURLConnection setupHttpConnection(byte[] incomingBinaryPOSTData,
+                                                  String serviceURL,
+                                                  JsonObject credentials) throws IOException {
+
+        String serviceUser = credentials.getString("username");
+        String servicePass = credentials.getString("password");
+        String auth = serviceUser + ":" + servicePass;
+
+        // Prepare the query string parameters.
+        // See the Speech to Text service documentation for details.
+        String queryString = "&model=en-US_NarrowbandModel"
+                + "&continuous=true";
+        if (DEBUG) {
+            System.out.println("POSTing [" + incomingBinaryPOSTData.length + "] bytes");
+            System.out.println("Using URL [" + serviceURL + API_PATH + "?" + queryString + "]");
+        }
+
+        // Prepare the HTTP connection to the service
+        HttpURLConnection conn = (HttpURLConnection) new URL(serviceURL + API_PATH + "?" + queryString).openConnection();
```

Comment:
This could be a function `openConnectionWithQueryString`, mostly to isolate that nasty casting.